### PR TITLE
Test Marker Class

### PR DESCRIPTION
**Pull request by NaamaHolzer**
Add tests for the Marker class:
- Test creation and initialization of an instance.
- Test marker -> entity conversion.
- Test entity -> marker conversion.

### CODE REVIEW

**Review by NoyFlat — APPROVED** [review]

File: capstone-project/src/main/java/com/google/sps/data/Marker.java [hunk 1]
```diff
@@ -136,8 +137,19 @@ public String getBlobKey() {
...
-    public void setBlobKey(String blobKey) {
-        this.blobKey = blobKey;
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Marker marker = (Marker) o;
+        return id == marker.id &&
+                Double.compare(marker.lat, lat) == 0 &&
+                Double.compare(marker.lng, lng) == 0 &&
```
Comment (orarch) [inline capstone-project/src/main/java/com/google/sps/data/Marker.java:147]: Why not ==?

File: capstone-project/src/main/javaTests/com/google/sps/data/MarkerTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+@RunWith(JUnit4.class)
+public final class MarkerTest {
+
+    private static Marker marker;
+    private static Entity markerEntity;
+    private static final LocalServiceTestHelper helper =
+            new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
```
Comment (orarch) [inline capstone-project/src/main/javaTests/com/google/sps/data/MarkerTest.java:26]: Why is this needed in this test class?

File: capstone-project/src/main/javaTests/com/google/sps/data/MarkerTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                .setLat(1)
+                .setLng(1)
+                .setAnimal("animal")
+                .setReporter("reporter")
+                .setDescription("description")
+                .setUserId(Optional.of("userId"))
+                .setBlobKey("blobKey")
+                .build();
+        helper.setUp();
+        markerEntity = new Entity("Marker");
```
Comment (orarch) [inline capstone-project/src/main/javaTests/com/google/sps/data/MarkerTest.java:41]: Try using:

new Entity("marker", 1111)

Then you don't need to sky on the entity

**Review by orarch — APPROVED** [review]
